dfget/core/uploader/uploader_util.go

```diff
@@ -47,6 +48,19 @@ func getTaskFile(taskFileName string) (*os.File, error) {
 	}
 
 	taskPath := helper.GetServiceFile(taskFileName, tc.dataDir)
+
+	// Pre-check the length of the file as expected.
+	fileInfo, err := os.Stat(taskPath)
+	if err != nil {
+		return nil, err
+	}
+	fileSize := fileInfo.Size()
+	fmt.Println(fileSize)
+	fmt.Printf("%+v", params)
+	if fileSize-params.start < params.pieceLen {
```

Comment:
The condition here is not sufficient, such as this case:
```
fileSize = 10
params.sart = -2
params.pieceLen = 11
```

Reply:
In fact, we will check the legality of range before that. 

https://github.com/dragonflyoss/Dragonfly/blob/master/dfget/core/uploader/uploader_util.go#L59